### PR TITLE
Extending with custom manual instrumentation

### PR DESCRIPTION
**Pull request by zeitlinger**
Fixes https://github.com/open-telemetry/opentelemetry.io/issues/5713

### CODE REVIEW

File: content/en/docs/zero-code/java/agent/annotations.md [hunk 1]
```diff
@@ -129,6 +129,5 @@ instrumented.
 ## Next steps
 
 Beyond the use of annotations, the OpenTelemetry API allows you to obtain a
-tracer that can be used for
-[Manual Instrumentation](/docs/languages/java/instrumentation/) and execute code
+tracer that can be used for [Manual Instrumentation](../api) and execute code
```
Comment (svrnm) [inline content/en/docs/zero-code/java/agent/annotations.md:132]:
```suggestion
tracer that can be used for [custom instrumentation](../api) and execute code
```

File: content/en/docs/zero-code/java/agent/api.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
+---
+title: Manual instrumentation
```
Comment (svrnm) [inline content/en/docs/zero-code/java/agent/api.md:2]:
```suggestion
title: Extending with custom manual instrumenation
linkTitle: Extending instrumentation
```

based on @jack-berg's title suggestion

File: content/en/docs/zero-code/java/spring-boot-starter/annotations.md [hunk 3]
```diff
@@ -120,3 +120,9 @@ annotation:
 | Name    | Type     | Description    | Default Value         |
 | ------- | -------- | -------------- | --------------------- |
 | `value` | `String` | Attribute name | Method parameter name |
+
+## Next steps
+
+Beyond the use of annotations, the OpenTelemetry API allows you to obtain a
+tracer that can be used for [Manual Instrumentation](../api) and execute code
```
Comment (svrnm) [inline content/en/docs/zero-code/java/spring-boot-starter/annotations.md:127]:
```suggestion
tracer that can be used for [custom instrumentation](../api) and execute code
```

File: content/en/docs/zero-code/java/spring-boot-starter/api.md [hunk 4]
```diff
@@ -0,0 +1,73 @@
+---
+title: Manual instrumentation
```
Comment (svrnm) [inline content/en/docs/zero-code/java/spring-boot-starter/api.md:2]:
```suggestion
title: Extending with custom manual instrumenation
linkTitle: Extending instrumentation
```

**Review by svrnm — CHANGES_REQUESTED** [review]
thanks

**Comment by zeitlinger** [comment]
@svrnm can you check?

**Review by svrnm — CHANGES_REQUESTED** [review]
One more thing, I am not sure if `api` is the right name for that, since we do not even mention "api" in the title, description, so either we do that, e.g. "Adding custom instrumentations with the API" or we rename the file to something like "extending-instrumentation.md" as well. wdyt?

**Comment by zeitlinger** [comment]
@svrnm please check again

File: content/en/docs/zero-code/java/agent/api.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
+---
+title: Extending with custom manual instrumentation using the OpenTelemetry API
+linkTitle: Extending instrumentation
+description: Extending a Java agent with custom manual instrumentation.
+weight: 21
+---
```
Comment (svrnm) [inline content/en/docs/zero-code/java/agent/api.md:6]: the title is now longer then the description, here is a proposal

```suggestion
---
title: Extending instrumentations with the API
linkTitle: Extend with the API
description: Use the OpenTelemetry API in combination with the Java agent to extend the automatically generated telemetry with custom instrumentations
weight: 21
---
```

@chalin @jack-berg can you take a look as well?

Comment (chalin) [reply content/en/docs/zero-code/java/agent/api.md:6]: Your proposal LGTM.

Comment (zeitlinger) [reply content/en/docs/zero-code/java/agent/api.md]: I adjusted slightly:

> Use the OpenTelemetry API in combination with the Java agent to extend the automatically generated telemetry with custom spans and metrics

**Comment by zeitlinger** [comment]
@svrnm can you check?

**Comment by svrnm** [comment]
> @svrnm can you check?

I did, but I'd like to get some feedback from @open-telemetry/java-approvers or @open-telemetry/java-instrumentation-approvers as well

**Comment by zeitlinger** [comment]
@jack-berg can you check?

File: content/en/docs/zero-code/java/spring-boot-starter/api.md [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+  public MyController(OpenTelemetry openTelemetry) {
+    this.tracer = openTelemetry.getTracer("application");
+  }
+}
+```
+
+Use the `Tracer` to create a span as explained in the
+[Span](/docs/languages/java/api/#span) section.
+
+A full example can be found [example repository].
```
Comment (trask) [inline content/en/docs/zero-code/java/spring-boot-starter/api.md:51]:
```suggestion
A full example can be found in the [example repository].
```

File: content/en/docs/zero-code/java/spring-boot-starter/api.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+  public MyController(OpenTelemetry openTelemetry) {
+    this.meter = openTelemetry.getMeter("application");
+  }
+}
+```
+
+Use the `Meter` to create a counter, gauge or histogram as explained in the
+[Meter](/docs/languages/java/api/#meter) section.
+
+A full example can be found [example repository].
```
Comment (trask) [inline content/en/docs/zero-code/java/spring-boot-starter/api.md:72]:
```suggestion
A full example can be found in the [example repository].
```

File: content/en/docs/zero-code/java/spring-boot-starter/annotations.md [hunk 8]
```diff
@@ -120,3 +120,9 @@ annotation:
 | Name    | Type     | Description    | Default Value         |
 | ------- | -------- | -------------- | --------------------- |
 | `value` | `String` | Attribute name | Method parameter name |
+
+## Next steps
+
+Beyond the use of annotations, the OpenTelemetry API allows you to obtain a
+tracer that can be used for [custom instrumentation](../api) and execute code
+within the scope of that span.
```
Comment (trask) [inline content/en/docs/zero-code/java/spring-boot-starter/annotations.md:128]: I didn't follow what "executing code within the scope of that span" meant, maybe reword or remove?

```suggestion
Beyond the use of annotations, the OpenTelemetry API allows you to obtain a
tracer that can be used [custom instrumentation](../api).
```

File: content/en/docs/zero-code/java/agent/annotations.md [hunk 9]
```diff
@@ -129,6 +129,5 @@ instrumented.
 ## Next steps
 
 Beyond the use of annotations, the OpenTelemetry API allows you to obtain a
-tracer that can be used for
-[Manual Instrumentation](/docs/languages/java/instrumentation/) and execute code
+tracer that can be used for [custom instrumentation](../api) and execute code
 within the scope of that span.
```
Comment (trask) [inline content/en/docs/zero-code/java/agent/annotations.md:133]: same question (though I see it was pre-existing text...)
```suggestion
tracer that can be used for [custom instrumentation](../api).
```

File: content/en/docs/zero-code/java/agent/api.md [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.trace.Tracer;
+
+Tracer tracer = GlobalOpenTelemetry.getTracer("application");
+```
+
+Use the `Tracer` to create a span as explained in the
+[Span](/docs/languages/java/api/#span) section.
+
+A full example can be found [example repository].
```
Comment (trask) [inline content/en/docs/zero-code/java/agent/api.md:67]:
```suggestion
A full example can be found in the [example repository].
```

File: content/en/docs/zero-code/java/agent/api.md [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.metrics.Meter;
+
+Meter meter = GlobalOpenTelemetry.getMeter("application");
+```
+
+Use the `Meter` to create a counter, gauge or histogram as explained in the
+[Meter](/docs/languages/java/api/#meter) section.
+
+A full example can be found [example repository].
```
Comment (trask) [inline content/en/docs/zero-code/java/agent/api.md:81]:
```suggestion
A full example can be found in the [example repository].
```

File: content/en/docs/languages/java/api.md [hunk 12]
```diff
@@ -600,7 +600,7 @@ public class OpenTelemetryUsage {
 ```
 <!-- prettier-ignore-end -->
 
-### GlobalOpenTelemetry
+#### GlobalOpenTelemetry
```
Comment (trask) [inline content/en/docs/languages/java/api.md:603]: I'm not sure about the organization of the additions in this file

Maybe could just add two small notes right above here, something like "If you are using the Java agent, see X for information about how to obtain an OpenTelemetry instance" (and similar note for Spring Boot starter)

**Review by trask — COMMENTED** [review]
I think the sections for the Java agent and Spring Boot starter are great additions.

I only have question about how to structure the addition to the general API page.

File: content/en/docs/languages/java/api.md (hunk 12 above)
Comment (jack-berg) [reply content/en/docs/languages/java/api.md:603]: I agree - I think an alert could be good. Something like
```
{{% alert title="Spring Boot Starter" %}} 
The Spring Boot starter is a special case where [...].
{{% /alert %}}
```

Comment (zeitlinger) [reply content/en/docs/languages/java/api.md:603]: I think it's useful if to have a hint it the ToC - which is not possible with an info box

image

Comment (trask) [reply content/en/docs/languages/java/api.md:603]: I think I prefer not having it in the TOC

Comment (zeitlinger) [reply content/en/docs/languages/java/api.md:603]: the use case I have in mind is this:

- user goes to https://opentelemetry.io/docs/languages/java/api/#counter (or any other instrument)
- user is wondering about this line: `public static void counterUsage(Meter meter) {` - where does `Meter` come from?

How would they know that they have to navigate to "OpenTelemetry"?

Comment (svrnm) [reply content/en/docs/languages/java/api.md:603]: I see where you are coming from, @zeitlinger, but I also see the issue that we should not overpopulate the TOC. A user using the agent / starter should have starter their reading on those pages, so ideally they have this information already. 

It is also easier to add something to the TOC later vs adding it now and removing it later, so let's start with the info boxes.

Comment (zeitlinger) [reply content/en/docs/languages/java/api.md]: thanks for the reviews - updated :smile:

**Review by svrnm — APPROVED** [review]

**Review by chalin — APPROVED** [review]
